src/bin/cargo.rs

```diff
@@ -205,8 +205,10 @@ fn execute_subcommand(config: &Config,
             }))
         }
     };
-    try!(util::process(&command).args(&args[1..]).exec());
-    Ok(())
+    match util::process(&command).args(&args[1..]).exec() {
+        Err(..) => Err(human(format!("Third party subcommand {:?} exited unsuccessfully.", command))),
```

Comment:
Fixed. Actually, now there is some duplication in error message, but if we suppose one is from 3rd party command and one from the cargo itself, it is probably fine.

    $ cargo check
       Compiling hexadecimal v0.0.0 (file:///home/user/projects/rust/hexadecimal)
    src/lib.rs:1:1: 1:3 error: expected item, found `ub`
    src/lib.rs:1 ub fn hex_to_int(s: &str) -> Option<usize> {
                 ^~
    error: aborting due to previous error
    Could not compile `hexadecimal`.

    To learn more, run the command again with --verbose.
    third party subcommand `cargo-check` exited unsuccessfully

    To learn more, run the command again with --verbose.